src/test/java/GmailClientTest.java

```diff
@@ -130,15 +140,54 @@ public void getQueryStringDontFilterForImportantTag() {
   }
 
   @Test
-  public void getQueryStringCombined() {
-    // Should return query matching all specified rules
-    // trailing spaces on sub-queries do not matter for this method, so they will not be considered
-    String multipleFilterQuery =
-        GmailClient.emailQueryString(ONE_UNIT_OF_TIME, DAYS_UNIT, true, true, SAMPLE_EMAIL);
-
-    Assert.assertTrue(multipleFilterQuery.contains(ONE_DAY_QUERY));
-    Assert.assertTrue(multipleFilterQuery.contains(UNREAD_EMAILS_QUERY));
-    Assert.assertTrue(multipleFilterQuery.contains(IMPORTANT_EMAILS_QUERY));
-    Assert.assertTrue(multipleFilterQuery.contains(FROM_EMAIL_QUERY));
+  public void getQueryStringNoSubjectWords() {
+    String subjectLineQuery = GmailClient.wordsInSubjectLineQuery(EMPTY_SUBJECT_LINE_WORDS);
+
+    Assert.assertEquals(subjectLineQuery, EMPTY_QUERY);
+  }
+
+  @Test
+  public void getQueryStringSomeSubjectWords() {
+    String subjectLineQuery = GmailClient.wordsInSubjectLineQuery(SOME_SUBJECT_LINE_WORDS);
+
+    // Query should be in the form of "subject:("wordOne" OR "wordTwo" OR "wordThree")"
+    // The order of the words in the brackets does not matter.
+    String actualSubjectLineQueryPrefix =
+        subjectLineQuery.substring(0, subjectLineQuery.indexOf(':') + 1);
+    List<String> actualSubjectLineWords =
+        Arrays.asList(
+            subjectLineQuery
+                .substring(subjectLineQuery.indexOf('(') + 1, subjectLineQuery.indexOf(')'))
+                .split(" OR "));
```

Comment:
You could use a regular expression (regex) to match the prefix and extract the words, all in a single method call.

Alternatively, if you want to assume the order is fixed, you could just match the exact string, or match a regex directly. (I don't think you are actually handling different word orders in this test, unless I am missing something?)

Reply:
Agreed - I'm just going to make the assumption here that the order of the words should be the same in the string (while they don't have to be, I don't think this is a bad assumption - there would be no reason to change the order) so I can compare the strings exactly.